### PR TITLE
Add Test for Header Component

### PR DESCRIPTION
**Pull request by Paul-Blankson**
This pull request adds comprehensive unit tests for the HeaderComponent in the frontend quiz application. The tests cover various functionalities, ensuring the component behaves as expected under different scenarios.

**Changes**

1. Component Creation Test

- Verifies that the HeaderComponent is created successfully.

2. Initialization Tests

- **isFirstPage Initialization:** Ensures isFirstPage is correctly initialized from QuizService.
- **Theme Initialization:** Checks that the theme is initialized from localStorage.

3. Theme Toggle Tests

- **Toggle Theme:** Verifies that the theme toggles correctly and updates localStorage.
- **Apply Theme Class:** Ensures the dark theme class is applied or removed based on the theme state.

4. Header Icon and Background Tests

- Verifies the correct header icon and background are set based on the selected subject from QuizService.

5. Logo Container Visibility Test

- Ensures the logo container is hidden when isFirstPage is true and displayed when false.

6. Subscription Cleanup Test

- Verifies that all subscriptions are unsubscribed when the component is destroyed.

7. Theme Toggle Checkbox Test

- Ensures the theme toggle checkbox reflects the correct theme state.

Please review these test cases to ensure they cover all edge cases and scenarios.

### CODE REVIEW

File: src/app/components/main-content/main-content.component.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ import { QuizData } from '../../types';
   styleUrl: './main-content.component.css',
 })
 export class MainContentComponent {
-  cards: QuizData[] = [];
-  isQuestionSection = false;
+ public cards: QuizData[] = [];
```
Comment (amt-augustineopokukesse) [inline src/app/components/main-content/main-content.component.ts:14]: why are you using public access modifiers?

**Review by amt-augustineopokukesse — APPROVED** [review]

File: src/app/components/main-content/main-content.component.ts (hunk 1 above)
Comment (Paul-Blankson) [reply src/app/components/main-content/main-content.component.ts:14]: Although in TypeScript properties are public by default, I have made a habit of declaring them as such, so it sticks with me and is considered part of my coding style.

I think this improves readability and consistency of code, especially in large teams or when switching between projects. However, I see your point, and I'm open to your suggestion on whether this is necessary or if it's better to rely on TypeScript's default behavior.
